File: frontend/src/types/project.ts
export interface Project {
  id: string;
  user_id: string;
  name: string;
  description: string | null;
  prompt: string;
  aspect_ratio: string;
  duration_seconds: number;
  status: ProjectStatus;
  brand_context_id: string | null;
  settings: ProjectSettings;
  quality: "draft" | "standard" | "high" | "ultra";
  output_format: string;
  output_fps: number;
  output_bitrate: string;
  progress: number;
  current_phase: string | null;
  error_message: string | null;
  generation_time_ms: number | null;
  render_time_ms: number | null;
  file_size_bytes: number | null;
  version: number;
  tags: string[];
  scenes?: Scene[];
  video_url?: string | null;
  thumbnail_url?: string | null;
  created_at: string;
  updated_at: string;
}

export type ProjectStatus = "draft" | "generating" | "rendering" | "ready" | "failed" | "archived";

export interface ProjectSettings {
  style: string;
  mood: string;
  pacing: string;
  colorGrading: string;
  transitions: string;
  textAnimation: string;
  musicGenre: string;
  voiceType: string;
  voiceGender: string;
}

export interface ProjectCreateInput {
  name: string;
  prompt: string;
  description?: string;
  aspect_ratio?: string;
  duration_seconds?: number;
  quality?: string;
  settings?: Partial<ProjectSettings>;
  tags?: string[];
}

export interface ProjectUpdateInput {
  name?: string;
  description?: string;
  prompt?: string;
  aspect_ratio?: string;
  duration_seconds?: number;
  quality?: string;
  settings?: Partial<ProjectSettings>;
  tags?: string[];
}

File: backend/app/schemas/project.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class ProjectCreateRequest(BaseModel):
    name: str
    description: str | None = None
    prompt: str
    aspect_ratio: str = "16:9"
    duration_seconds: int = 30
    quality: str = "high"
    settings: dict = {}
    tags: list[str] = []


class ProjectUpdateRequest(BaseModel):
    name: str | None = None
    description: str | None = None
    prompt: str | None = None
    aspect_ratio: str | None = None
    duration_seconds: int | None = None
    quality: str | None = None
    settings: dict | None = None
    tags: list[str] | None = None


class ProjectResponse(BaseModel):
    id: uuid.UUID
    name: str
    description: str | None
    prompt: str
    aspect_ratio: str
    duration_seconds: int
    status: str
    progress: float
    quality: str
    output_format: str
    version: int
    tags: list[str]
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ProjectListResponse(BaseModel):
    id: uuid.UUID
    name: str
    description: str | None
    status: str
    progress: float
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
